### ISSUE TITLE
clonalDiversity plots grouped by patient for each type from vignette data

### ISSUE THREAD

**Issue by pm-Genome2021** [BorchLab/scRepertoire#504]
Hi,

I am trying to create a clonalDiversity plot where I display the diversity metrics for each patient grouped according to the 'Type' in the x.axis from the vignette data from the page below:

https://www.borch.dev/uploads/screpertoire/articles/clonal_diversity

I use the following command: 

clonalDiversity(combined.TCR, 
                cloneCall = "gene",group.by = "Patient", 
                x.axis = "Type")

However, in the plot, I only see one of the types ('B') instead of both the types. 

![Image](https://github.com/user-attachments/assets/272b105e-46ac-4e1a-9548-3bf37ea7cfe5)

**sessionInfo()**

R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Rocky Linux 9.5 (Blue Onyx)

Matrix products: default
BLAS/LAPACK: /data/apps/software/spack/linux-rocky9-x86_64_v3/gcc-11.3.1/openblas-0.3.28-bfaxlal4muj4t6q26ahjufgpkqvt4pvu/lib/libopenblas-r0.3.28.so;  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_1.0.0      scRepertoire_2.5.2 ggplot2_3.5.1     

loaded via a namespace (and not attached):
  [1] gridExtra_2.3               rlang_1.1.6                
  [3] magrittr_2.0.3              matrixStats_1.5.0          
  [5] compiler_4.4.1              png_0.1-8                  
  [7] vctrs_0.6.5                 reshape2_1.4.4             
  [9] ggalluvial_0.12.5           gsl_2.1-8                  
 [11] quantreg_5.99.1             rvest_1.0.4                
 [13] stringr_1.5.1               pkgconfig_2.0.3            
 [15] crayon_1.5.3                fastmap_1.2.0              
 [17] XVector_0.46.0              labeling_0.4.3             
 [19] ggraph_2.2.1                UCSC.utils_1.2.0           
 [21] MatrixModels_0.5-3          purrr_1.0.4                
 [23] zlibbioc_1.52.0             cachem_1.1.0               
 [25] GenomeInfoDb_1.42.1         jsonlite_2.0.0             
 [27] DelayedArray_0.32.0         tweenr_2.0.3               
 [29] VGAM_1.1-12                 parallel_4.4.1             
 [31] tensorflow_2.16.0           R6_2.6.1                   
 [33] stringi_1.8.7               reticulate_1.42.0          
 [35] parallelly_1.41.0           GenomicRanges_1.58.0       
 [37] Rcpp_1.0.14                 assertthat_0.2.1           
 [39] SummarizedExperiment_1.36.0 future.apply_1.11.3        
 [41] iNEXT_3.0.1                 base64enc_0.1-3            
 [43] IRanges_2.40.1              Matrix_1.7-2               
 [45] splines_4.4.1               igraph_2.1.4               
 [47] tidyselect_1.2.1            stringdist_0.9.15          
 [49] abind_1.4-8                 viridis_0.6.5              
 [51] codetools_0.2-20            listenv_0.9.1              
 [53] lattice_0.22-6              tibble_3.3.0               
 [55] plyr_1.8.9                  Biobase_2.66.0             
 [57] withr_3.0.2                 survival_3.8-3             
 [59] future_1.34.0               polyclip_1.10-7            
 [61] xml2_1.3.8                  pillar_1.10.2              
 [63] MatrixGenerics_1.18.1       whisker_0.4.1              
 [65] stats4_4.4.1                generics_0.1.3             
 [67] sp_2.1-4                    S4Vectors_0.44.0           
 [69] munsell_0.5.1               scales_1.3.0               
 [71] globals_0.16.3              immApex_1.0.5              
 [73] glue_1.8.0                  cubature_2.1.1             
 [75] tools_4.4.1                 SparseM_1.84-2             
 [77] graphlayouts_1.2.2          dotCall64_1.2              
 [79] tidygraph_1.3.1             grid_4.4.1                 
 [81] tidyr_1.3.1                 evmix_2.12                 
 [83] truncdist_1.0-2             dotty_0.1.0                
 [85] keras3_1.4.0                colorspace_2.1-1           
 [87] SingleCellExperiment_1.28.1 GenomeInfoDbData_1.2.13    
 [89] ggforce_0.4.2               cli_3.6.3                  
 [91] evd_2.3-7.1                 tfruns_1.5.3               
 [93] spam_2.11-1                 S4Arrays_1.6.0             
 [95] viridisLite_0.4.2           ggdendro_0.2.0             
 [97] dplyr_1.1.4                 gtable_0.3.6               
 [99] hash_3.0.1                  zeallot_0.2.0              
[101] digest_0.6.37               progressr_0.15.1           
[103] BiocGenerics_0.52.0         SparseArray_1.6.1          
[105] ggrepel_0.9.6               rjson_0.2.23               
[107] SeuratObject_5.0.2          farver_2.1.2               
[109] memoise_2.0.1               lifecycle_1.0.4            
[111] httr_1.4.7                  MASS_7.3-64  

Can you please help with this issue?

**Comment by ncborcherding** [comment]
@pm-Genome2021 

Thanks for reaching out - this is actually a bug. x.axis is not being fully integrated into the bootstrap calculations. I am working on some functional overhauls including clonalDiversity() and will get this patched next week. 

Nick

**Comment by pm-Genome2021** [comment]
Thanks a lot for this quick fix and for this very useful package.
